### PR TITLE
Use cached instead of lru

### PR DESCRIPTION
**Pull request by bowenwang1996**
Change the lru crate used in storage from lru to cached to make lru usage consistent across different crates.

### CODE REVIEW

File: core/storage/src/storages/mod.rs [hunk 1]
```diff
@@ -307,35 +307,39 @@ fn extend_with_cache<T: Clone + Encode>(
...
-        Ok(Some(value))
-    } else if let Some(data) = storage.get(Some(col), key)? {
-        let result = Decode::decode(data.as_ref())?;
-        cache.put(key_vec.clone(), result);
-        Ok(cache.get(&key_vec))
-    } else {
-        Ok(None)
+    // This code block is safe. Has to use unsafe due to Cached API limitations,
+    // mainly that there is not lookup method that takes &self.
+    unsafe {
```
Comment (nearmax) [inline core/storage/src/storages/mod.rs]: This should be resolvable without unsafe. Here is the minimum reproducible example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b0bebdb3467a12d99b1b219dce688b05 I asked in Rust discord why Rust NLL does not resolve this.

Comment (nearmax) [reply core/storage/src/storages/mod.rs]: Here is my suboptimal solution with double lookup. I suggest we go with it instead, unless we later determine that it creates a performance bottleneck.
```
    let key_vec = key.to_vec();
    if cache.cache_get(&key_vec).is_some() {
        return Ok(Some(cache.cache_get(&key_vec).unwrap()));
    }
    if let Some(data) = storage.get(Some(col), key)? {
        let result = Decode::decode(data.as_ref())?;

        let key_vec = key_vec.to_vec();
        cache.cache_set(key_vec.to_vec(), result);
        return Ok(cache.cache_get(&key_vec));
    }
    Ok(None)
}
```

Comment (nearmax) [reply core/storage/src/storages/mod.rs]: Rust gurus answered -- it is the current limitation of compiler. I suggest we go with suboptimal oppose to unsafe.

Comment (bowenwang1996) [reply core/storage/src/storages/mod.rs]: I don't think not using unsafe is possible given that `cache_get` takes `&mut self` instead of `&self`.

Comment (nearmax) [reply core/storage/src/storages/mod.rs]: The above code that I pasted is working.

Comment (bowenwang1996) [reply core/storage/src/storages/mod.rs]: I see. I misread.

**Pull request merged**